nodewatcher/nodewatcher.py

```diff
@@ -156,19 +160,61 @@ def _maintain_size(asg_name, asg_client):
     :param asg_client: ASG boto3 client
     :return: True if the desired capacity is lower than the configured min size.
     """
-    asg = asg_client.describe_auto_scaling_groups(AutoScalingGroupNames=[asg_name]).get("AutoScalingGroups")[0]
-    _capacity = asg.get("DesiredCapacity")
-    _min_size = asg.get("MinSize")
-    log.info("DesiredCapacity is %d, MinSize is %d" % (_capacity, _min_size))
-    if _capacity > _min_size:
-        log.debug("Capacity greater than min size.")
+    try:
+        asg = asg_client.describe_auto_scaling_groups(AutoScalingGroupNames=[asg_name]).get("AutoScalingGroups")[0]
+        _capacity = asg.get("DesiredCapacity")
+        _min_size = asg.get("MinSize")
+        log.info("DesiredCapacity is %d, MinSize is %d" % (_capacity, _min_size))
+        if _capacity > _min_size:
+            log.debug("Capacity greater than min size.")
+            return False
+        else:
+            log.debug("Capacity less than or equal to min size.")
+            return True
+    except Exception as e:
+        log.error(
+            "Failed when checking min cluster size with exception %s. Assuming capacity is greater than min size.", e
+        )
         return False
-    else:
-        log.debug("Capacity less than or equal to min size.")
-        return True
 
 
-@retry(wait_fixed=10000, retry_on_result=lambda result: result is False)
+def _dump_logs(instance_id):
+    """Dump gzipped /var/log dir to /home/logs/compute/$instance_id.tar.gz"""
+    logs_dir = "/home/logs/compute"
+    filename = "{0}/{1}.tar.gz".format(logs_dir, instance_id)
+    try:
+        try:
+            os.makedirs(logs_dir)
+        except OSError as e:
```

Comment:
Instead of having a double nested `try/except` block, use two except clauses:

```
try:
...
except e1:
...
except e2:
...
```

Reply:
this first try..except is to handle the case where the log dir already exists. It is a bit ugly but unfortunately they added a flag to the makedirs method only in later python versions. I can't merge the except cause I still want to dump the files if the exception is due to existing dir.

Reply:
I still don't see it, how about:

```python
try:
    os.makedirs(logs_dir)
except OSError as e:
if e.errno == errno.EEXIST:
    log.info("Dumping logs to %s", filename)
    with tarfile.open(filename, "w|gz") as archive:
        archive.add("/var/log", recursive=True)
except Exception as e:
    log.warning("Failed while dumping logs to %s with exception %s.", filename, e)
```

Seems like that should do the trick. Am I missing something?

Reply:
the tarfile needs to be created also when os.makedirs succeeds since makedirs is creating the log dir and not the file